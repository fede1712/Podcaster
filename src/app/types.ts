export interface Podcast {
  "im:name": { label: string };
  "im:image": image[];
  summary: { label: string };
  "im:price": { label: string; attributes: { amount: string; currency: string } };
  "im:contentType": { attributes: { term: string; label: string } };
  rights: { label: string };
  title: { label: string };
  link: { attributes: { rel: string; type: string; href: string } };
  id: { label: string; attributes: { "im:id": string } };
  "im:artist": { label: string; attributes: { href: string } };
  category: { attributes: { "im:id": string; term: string; scheme: string; label: string } };
  "im:releaseDate": { label: string; attributes: { label: string } };
}

export type image = { label: string; attributes: { height: string } };

export interface Episode {
  country: string;
  artworkUrl160: string;
  episodeFileExtension: string;
  episodeContentType: string;
  artworkUrl600: string;
  collectionViewUrl: string;
  trackTimeMillis: number;
  shortDescription: string;
  closedCaptioning: string;
  collectionId: number;
  collectionName: string;
  genres: genres[];
  episodeGuid: string;
  description: string;
  releaseDate: string;
  previewUrl: string;
  trackId: number;
  trackName: string;
  feedUrl: string;
  artistIds: [];
  episodeUrl: string;
  artworkUrl60: string;
  contentAdvisoryRating: string;
  trackViewUrl: string;
  kind: string;
  wrapperType: string;
  genreIds: string[];
  primaryGenreName: string;
  currency: string;
  trackCount: number;
  trackExplicitness: string;
  artistName: string;
  collectionCensoredName: string;
  trackCensoredName: string;
  artworkUrl30: string;
  artworkUrl100: string;
  collectionPrice: number;
  trackPrice: number;
  collectionHdPrice: number;
  collectionExplicitness: string;
}

export type genres = { name: string; id: string };
