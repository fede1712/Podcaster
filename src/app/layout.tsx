import type { Metadata } from "next";
import { Inter } from "next/font/google";
import "./globals.css";
import Link from "next/link";
import { Header } from "./ui/header/header";

export const metadata: Metadata = {
  title: "Podcaster",
  description: "Itunes top podcasts",
};

export default function RootLayout({
  children,
}: Readonly<{
  children: React.ReactNode;
}>) {
  return (
    <html lang="en">
      <body className="m-8">
        <Header />
        {children}
      </body>
    </html>
  );
}
