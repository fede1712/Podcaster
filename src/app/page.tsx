import { getPodcasts } from "./lib/data";
import { PodcastCard } from "./ui/podcastCard/podcastCard";

export default async function Home() {
  const data = await getPodcasts();
  console.log(data.feed.entry[1]);

  return (
    <div className="grid grid-cols-4 gap-14 m-14">
      {data.feed.entry.map((podcast: any) => {
        return <PodcastCard author={podcast["im:artist"].label} title={podcast["im:name"].label} src={podcast["im:image"][2].label} />;
      })}
    </div>
  );
}
