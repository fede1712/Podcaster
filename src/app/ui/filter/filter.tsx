import { Podcast } from "@/app/types";

export const Filter = ({
  data,
  results,
  setResults,
  setSearch,
  search,
}: {
  data: Podcast[];
  results: Podcast[];
  setResults: (result: Podcast[]) => void;
  setSearch: (search: string) => void;
  search: string;
}) => {
  const handleSearch = (value: string) => {
    setSearch(value);
    if (search.length > 0) {
      const filteredResults = data.filter((podcast: Podcast) => {
        if (
          podcast["im:name"].label.toLowerCase().includes(value.toLowerCase()) ||
          podcast["im:artist"].label.toLowerCase().includes(value.toLowerCase())
        ) {
          return true;
        } else {
          return false;
        }
      });
      setResults(filteredResults);
    }
  };

  return (
    <div className="flex justify-end m-14" data-testId="filter">
      <div className="bg-cyan-600 rounded-md p-2 m-2">
        <p className="text-white">{results.length}</p>
      </div>
      <input type="text" placeholder="Filter podcast..." className="p-2 border rounded-md" onChange={(e) => handleSearch(e.target.value)} />
    </div>
  );
};
