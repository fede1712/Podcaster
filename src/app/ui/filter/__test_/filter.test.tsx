import React from "react";
import { render, screen } from "@testing-library/react";
import { Filter } from "../filter";
import { Podcast } from "@/app/types";

test("Renders filter correctly", () => {
  const podcast: Podcast[] = [
    {
      "im:name": { label: "Test podcast 1" },
      "im:image": [{ label: "http://example.com/artwork.jpg-1" }],
      summary: { label: "Summary podcast 1" },
      title: { label: "Title podcast 1" },
      id: { attributes: { "im:id": "1" } },
      "im:artist": { label: "Artist Podacast 1" },
    },
    {
      "im:name": { label: "Test podcast 2" },
      "im:image": [{ label: "http://example.com/artwork.jpg-2" }],
      summary: { label: "Summary podcast 2" },
      title: { label: "Title podcast 2" },
      id: { attributes: { "im:id": "2" } },
      "im:artist": { label: "Artist Podacast 2" },
    },
    {
      "im:name": { label: "Test podcast 3" },
      "im:image": [{ label: "http://example.com/artwork.jpg-3" }],
      summary: { label: "Summary podcast 3" },
      title: { label: "Title podcast 3" },
      id: { attributes: { "im:id": "3" } },
      "im:artist": { label: "Artist Podacast 3" },
    },
  ];

  const podcastResults: Podcast[] = [
    {
      "im:name": { label: "Test podcast 1" },
      "im:image": [{ label: "http://example.com/artwork.jpg-1" }],
      summary: { label: "Summary podcast 1" },
      title: { label: "Title podcast 1" },
      id: { attributes: { "im:id": "1" } },
      "im:artist": { label: "Artist Podacast 1" },
    },
  ];

  render(<Filter data={podcast} results={podcastResults} search="" setResults={() => jest.fn()} setSearch={() => jest.fn()} />);

  const results = screen.queryByText("1");
  expect(results).not.toBeNull();

  const { asFragment } = render(
    <Filter data={podcast} results={podcastResults} search="" setResults={() => jest.fn()} setSearch={() => jest.fn()} />
  );

  expect(asFragment()).toMatchSnapshot();
});
