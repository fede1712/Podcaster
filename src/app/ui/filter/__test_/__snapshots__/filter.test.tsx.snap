// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders filter correctly 1`] = `
<DocumentFragment>
  <div
    class="flex justify-end m-14"
  >
    <div
      class="bg-cyan-600 rounded-md p-2 m-2"
    >
      <p
        class="text-white"
      >
        1
      </p>
    </div>
    <input
      class="p-2 border rounded-md"
      placeholder="Filter podcast..."
      type="text"
    />
  </div>
</DocumentFragment>
`;
